Xamarin.Forms.Platform.iOS/Renderers/EntryRenderer.cs

```diff
@@ -488,9 +491,16 @@ void SetSelectionLengthFromRenderer(int selectionLength)
 			}
 		}
 
-        void UpdateIsReadOnly()
-        {
-            Control.UserInteractionEnabled = !Element.IsReadOnly;
-        }
-    }
+		void UpdateIsReadOnly()
+		{
+			Control.UserInteractionEnabled = !Element.IsReadOnly;
+		}
+
+		void UpdateClearButtonVisibility(bool forceUpdate)
```

Comment:
Can you explain why we need this `forceUpdate` flag?

Reply:
Sure, take a look [here](https://github.com/xamarin/Xamarin.Forms/pull/5277#discussion_r257570333)

Reply:
What do you think about below (make `UpdateClearButtonVisiblilty()` simple) ? 


```cs
protected override void OnElementChanged(ElementChangedEventArgs<Entry> e)
{
 //...
    UpdateCursorColor();
    UpdateIsReadOnly();
    if (Element.ClearButtonMode == InputViewClearButtonMode.WhileEditing) // or if (Element.IsSet(Entry.ClearButtonModeProperty))
        UpdateClearButtonVisibility();
// ...
}

void UpdateClearButtonVisibility()
{
    Control.ClearButtonMode =  Element.ClearButtonMode == InputViewClearButtonMode.WhileEditing ? UITextFieldViewMode.WhileEditing : UITextFieldViewMode.Never;
}
```

Reply:
This should be the preferred approach, imo. Setting Control.ClearButtonMode to its default value during initialization should not cost a performance hit, so there is no need to introduce an extra flag.

Reply:
@rookiejava Why would calling the `ClearButtonMode` bindable property on `ElementChanged` be better? A simple flag is much much less costly than reading bindable property.

Reply:
This is simply why `ClearButtonMode bindable property is always used. (regardless of flag)

```cs
bool showClearButton = Element.ClearButtonMode == InputViewClearButtonMode.WhileEditing;  //<--- this line is always executed.
```

Reply:
Maybe I wasn't clear. `Element.ClearButtonMode` actually calls `BindableObject.GetValue(ClearButtonModeProperty)` under the hood, which goes into BindableObject implementation and tries get the bindable property value. 
Doing this on each and every Entry initialization on a Page can be avoided by a simple flag per my implementation above.

Reply:
The flag you are talking (`forceUpdate`) is used to determine whether to call `Control.ClearButtonMode`, not `Element.ClearButtonMode`. In other words, regardless of whether the flag (`forceUpdate`) value is `true` or `false`, `Element.ClearButtonMode` is always called in your suggestion.

Reply:
Guys, thanks for your feedback, I implemented the changes. I removed the flags. 
I initially didn't like the idea of adding any more logic right in the `OnElementChanged()`, there's already quite some logic already. But on a 2nd thought I guess it's better to do that instead of having the logic of those flags...